OOP/HW_1.py

```diff
@@ -0,0 +1,546 @@
+# 
+# Напишите несколько примеров использования кода, которое показывает взаимодействие между объектами.
+# 
+# В тексте задания намерено не указано, какие аргументы должны принимать методы, пускай вам в этом поможет здравый смысл)
+# 
+# В этом задании не стоит флексить всякими продвинутыми штуками, для этого есть последующие
+# 
+# В этом задании можно использовать только модуль `datetime`
+
+# In[1]:
+
+
+from datetime import datetime
+
+
+# In[2]:
+
+
+class Chat:
+    def __init__(self, history=None):
+        if history is None:
+            history = []
+        self.chat_history = history
+        
+    def show_last_message(self):
+        print(self.chat_history[0].user, self.chat_history[0].text, 
+              self.chat_history[0].datetime, sep='\n')
+    
+    def get_history_from_time_period(self, begin=None, end=None):
+        if begin is None:
+            begin = self.chat_history[-1].datetime
+        if end is None:
+            end = self.chat_history[0].datetime
+        history = list(mes for mes in self.chat_history if mes.datetime >= begin and mes.datetime <= end)
+        return Chat(history)
+    
+    def show_chat(self):
+        for mes in self.chat_history[::-1]:
+            print(mes.user, mes.text, mes.datetime, sep='\n', end='\n❤️️❤️️❤️️\n')
+            
+    
+    def receive(self, message):
+        self.chat_history = [message] + self.chat_history
+
+
+class Message:
+    def __init__(self, text, user):
+        self.text = text
+        self.datetime = None
+        self.user = user.nickname
+    
+    def show(self):
+        print(self.user, self.text, self.datetime, sep='\n')
+        
+    def send(self, chat):
+        self.datetime = datetime.now()
+        chat.receive(self)
+        
+
+class User:
+    def __init__(self, name, surname, nickname):
+        self.name = name
+        self.surname = surname
+        self.nickname = nickname
+
+
+# In[3]:
+
+
+roma = User('Roma', 'Kruglikov', 'sunny')
+natasha = User('Natasha', 'Khotkina', 'crying')
+
+
+# In[4]:
+
+
+mes1 = Message('Hi, Natasha!', roma)
+mes2 = Message('Hi!', natasha)
+
+
+# In[5]:
+
+
+mes1.show()
+
+
+# In[6]:
+
+
+friends_chat = Chat()
+
+
+# In[7]:
+
+
+mes1.send(friends_chat)
+
+
+# In[8]:
+
+
+mes2.send(friends_chat)
+
+
+# In[9]:
+
+
+friends_chat.show_chat()
+
+
+# In[12]:
+
+
+chat_hist = friends_chat.get_history_from_time_period(begin=datetime(2023,2,25,18,46,59))
+
+
+# In[13]:
+
+
+chat_hist.show_chat()
+
+
+# # Задание 2 (3 балла)
+
+# В питоне как-то слишком типично и неинтересно происходят вызовы функций. Напишите класс `Args`, который будет хранить в себе аргументы, а функции можно будет вызывать при помощи следующего синтаксиса.
+# 
+# Использовать любые модули **нельзя**, да и вряд-ли это как-то поможет)
+
+# In[14]:
+
+
+class Args():
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+    def __rlshift__(self, func):
```

Comment:
![image](https://user-images.githubusercontent.com/48160953/226219642-0aa0bda8-e126-458f-9346-c9d720258d72.png)


```suggestion
        self.kwargs = kwargs
        
    def __rlshift__(self, func):
```